src/model-helpers/User.js

```diff
@@ -0,0 +1,173 @@
+  const accessObject = {
+    read: true,
+    append: true,
+    write: true,
+    control: true
+  };
+  const publicAccessObject = {
+    read: true,
+    append: false,
+    write: false,
+    control: false
+  };
+  let activityDataset;
+
+  const updateAndSaveActivity = async () => {
+    const activity = buildThing(createThing({ name: 'active' }))
+      .addDatetime(RDF_PREDICATES.dateModified, new Date())
+      .build();
+
+    activityDataset = setThing(activityDataset, activity);
+
+    await saveSolidDatasetAt(activityDocUrl, activityDataset, {
+      fetch: session.fetch
+    });
+  };
+
+  try {
+    activityDataset = await getSolidDataset(activityDocUrl, { fetch: session.fetch });
+    await updateAndSaveActivity();
+  } catch {
+    activityDataset = createSolidDataset();
+    await updateAndSaveActivity();
+
+    await setDocAclForUser(
+      session,
+      activityDocUrl,
+      'create',
+      session.info.webId,
+      accessObject,
+      publicAccessObject
+    );
+  }
+};
+
+/**
+ * Fetches the last time a user was active in pass
+ *
+ * @memberof User
+ * @function loadUserActivity
+ * @param {URL} podUrl - Url of user's pod to fetch from
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @returns {string} last time the user was active in PASS
+ */
+const loadUserActivity = async (podUrl, session) => {
+  const activityUrl = `${podUrl}public/active.ttl`;
+  try {
+    const solidDataset = await getSolidDataset(activityUrl, {
+      fetch: session.fetch
+    });
+    const activeThing = getThingAll(solidDataset)[0];
+    return getDatetime(activeThing, RDF_PREDICATES.dateModified);
+  } catch {
+    return null;
+  }
+};
+
+/**
+ * Creates a user object from a provided form submission
+ *
+ * @memberof User
+ * @function createUser
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {object} userSubmission - an object from a form submission containing the user creation data
+ * @returns {Promise} Promise - Updates last active time of user to lastActive.ttl
+ */
+export const createUser = async (session, userSubmission) => {
+  const { familyName, username, givenName, webId } = userSubmission;
+
+  let newUserPodUrl = null;
+  try {
+    [newUserPodUrl] = await getPodUrlAll(webId);
+  } catch {
+    [newUserPodUrl] = webId.split('profile');
+  }
+  newUserPodUrl = newUserPodUrl || webId.split('profile')[0];
+
+  const dateModified = await loadUserActivity(newUserPodUrl, session);
+
+  return {
+    familyName,
+    username,
+    givenName,
+    webId,
+    dateModified,
+    podUrl: newUserPodUrl
+  };
+};
+
+/**
+ * Converts an inrupt Thing taken from the users list into a user object
+ *
+ * @memberof User
+ * @function parseUserFromThing
+ * @param {import('@inrupt/solid-client').Thing} userThing - the Thing to build the user from
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @returns {object} user object
```

Comment:
This could be `{Promise<object>}` or `{Promise<userListObject>}` once we've updated the type definition for userListObject in typedefs.js.

Most of a self-note: Might need to update the naming for Types too though in a future PRs, like use `UserListObject` instead with PascalCase to make it more TypeScript-like.